supervisor/homeassistant/module.py

```diff
@@ -238,6 +241,12 @@ async def load(self) -> None:
         """Prepare Home Assistant object."""
         await asyncio.wait([self.secrets.load(), self.core.load()])
 
+        # Register for events
+        self.sys_bus.register_event(BusEvent.HARDWARE_NEW_DEVICE, self._hardware_events)
+        self.sys_bus.register_event(
+            BusEvent.HARDWARE_REMOVE_DEVICE, self._hardware_events
```

Comment:
The core now only seems to check for add, and not remove, should we follow that? <https://github.com/home-assistant/core/pull/55570/files#diff-6d2d912b8a4820fb22dea34720b0b6181699da8e51db2f24b45473bfb7e2cc60R148>

Reply:
None of the discoveries (zeroconf, ssdp, usb) support remove/unsee but likely will be at a later date.  Still we probably shouldn't rescan on removal right now.